x-pack/plugins/ecs_data_quality_dashboard/server/routes/results/get_results.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IRouter, Logger } from '@kbn/core/server';
+
+import { RESULTS_ROUTE_PATH, INTERNAL_API_VERSION } from '../../../common/constants';
+import { buildResponse } from '../../lib/build_response';
+import { buildRouteValidation } from '../../schemas/common';
+import { GetResultQuery } from '../../schemas/result';
+import type { Result, ResultDocument } from '../../schemas/result';
+import { API_DEFAULT_ERROR_MESSAGE } from '../../translations';
+import type { DataQualityDashboardRequestHandlerContext } from '../../types';
+import { createResultFromDocument } from './parser';
+import { API_RESULTS_INDEX_NOT_AVAILABLE } from './translations';
+
+export const getQuery = (patterns: string[]) => ({
+  size: 0,
+  query: {
+    bool: { filter: [{ terms: { 'rollup.pattern': patterns } }] },
+  },
+  aggs: {
+    latest: {
+      terms: { field: 'rollup.pattern', size: 10000 }, // big enough to get all patterns, but under `index.max_terms_count` (default 65536)
+      aggs: { latest_doc: { top_hits: { size: 1, sort: [{ '@timestamp': { order: 'desc' } }] } } },
+    },
+  },
+});
+
+export interface LatestAggResponseBucket {
+  key: string;
+  latest_doc: { hits: { hits: Array<{ _source: ResultDocument }> } };
+}
+
+export const getResultsRoute = (
+  router: IRouter<DataQualityDashboardRequestHandlerContext>,
+  logger: Logger
+) => {
+  router.versioned
+    .get({
+      path: RESULTS_ROUTE_PATH,
+      access: 'internal',
+      options: { tags: ['access:securitySolution'] },
+    })
+    .addVersion(
+      {
+        version: INTERNAL_API_VERSION,
+        validate: { request: { query: buildRouteValidation(GetResultQuery) } },
+      },
+      async (context, request, response) => {
+        const services = await context.resolve(['core', 'dataQualityDashboard']);
+        const resp = buildResponse(response);
+
+        let index: string;
+        try {
+          index = await services.dataQualityDashboard.getResultsIndexName();
+        } catch (err) {
+          logger.error(`[GET results] Error retrieving results index name: ${err.message}`);
+          return resp.error({
+            body: `${API_RESULTS_INDEX_NOT_AVAILABLE}: ${err.message}`,
+            statusCode: 503,
+          });
+        }
+
+        try {
+          // Confirm user has authorization for the requested patterns
+          const { patterns } = request.query;
+          const userEsClient = services.core.elasticsearch.client.asCurrentUser;
+          const privileges = await userEsClient.security.hasPrivileges({
+            index: [
+              { names: patterns.split(','), privileges: ['all', 'read', 'view_index_metadata'] },
```

Comment:
Consider revising the implementation of this `privileges` check, because per the following screenshot:

![data_quality_with_has_viewer_role](https://github.com/elastic/kibana/assets/4459398/3cfc3e27-d25d-453b-aca2-88b41ae9b66a)

the following privileges are required to check an index:

- `monitor` or `manage`
- `view_index_metadata`
- `read`

Consider:
- The `all` privilege is not required
- _either_ `monitor` or `manage` is required to check an index
  - In a future release, `read` may be expanded to be `read` or `read_cross_cluster`, if the Elasticsearch `stats` and `_ilm/explain` APIs are enhanced to support CCS.